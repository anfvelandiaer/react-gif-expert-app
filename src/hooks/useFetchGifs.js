import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";


export const useFetchGifs = ( category ) => {
    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    // EL segundo parÃ¡metro dice que se lanza el effect si la categoria cambia, 
    //si se deja un array vacio solo se ejecuta una vez, si se deja undefined se ejcuta siempre
    useEffect(()=> {
        getGifs(category)
        .then(imgs=> {
            setState( {
                data: imgs,
                loading: false
            });
        });
    }, [category]);

    return state;
}