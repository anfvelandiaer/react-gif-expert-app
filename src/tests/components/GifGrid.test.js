import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

// Simula la respuesta del useFetchGifs
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en gif grid", () => {
  const category = "One Punch";

  test("should mostrar el componente correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

     expect(wrapper).toMatchSnapshot();


  });

  test("debe mostrar items cuando se cargan imagenes useFecth", () => {
    const gifs = [
      {
        id: "ABC",
        URL: "HTTPS://localhost",
        title: "Cualquier Cosa",
      },
      {
        id: "123",
        URL: "HTTPS://localhost",
        title: "Cualquier Cosa",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

     // Como ya no est√° cargando no debe mostrar el mensaje del p
     expect(wrapper.find('p').exists()).toBe(false)
     // Probar que se crea la cantidad correcta
     expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

     // expect(wrapper).toMatchSnapshot();
  });
});
