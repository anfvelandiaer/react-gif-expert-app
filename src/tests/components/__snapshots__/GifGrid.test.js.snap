// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gif grid debe mostrar items cuando se cargan imagenes useFecth 1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      URL="HTTPS://localhost"
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en gif grid should mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
